bug: TFSec crashes pipeline with error referring to the default_action_deny check when run against an Azure Storage Account

**Describe the bug**
When run against an Azure Storage Account, TFSec crashes pipeline with error referring to the default_action_deny check when   network_rules - default_action is set as a variable in an env.hcl file when using Terragrunt for the build process.

**To Reproduce**
Steps to reproduce the behavior:
Run TFSec v1.19.1 against an Azure storage account. We are running in a pipeline using `/root/go/bin/tfsec --force-all-dirs`

Error detailed below is displayed.
This is occurring even after adding an ignore comment for the check:

```
#tfsec:ignore:azure-storage-default-action-deny
```

**Expected behavior**
TFSec runs and checks storage account resources and returns issues if present.

**Screenshots/Output**
```
Error: scan failed: job failed: runtime error: invalid memory address or nil pointer dereference
goroutine 187 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*infraRuleJob).Run.func1()
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/pkg/scanners/terraform/executor/pool.go:128 +0x45
panic({0x11fe580, 0x203c560})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/aquasecurity/defsec/internal/rules/azure/storage.glob..func2(0xc000991c00)
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/internal/rules/azure/storage/default_action_deny.go:37 +0x174
github.com/aquasecurity/defsec/internal/rules.RegisteredRule.Evaluate({0xab, {{0x1375473, 0xc}, {0x0, 0x0}, {0x138263f, 0x13}, {0x13e3287, 0x49}, {0x13f7a22, ...}, ...}, ...}, ...)
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/internal/rules/register.go:26 +0x3b
github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*infraRuleJob).Run(0x1340380)
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/pkg/scanners/terraform/executor/pool.go:132 +0xf9
github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*Worker).Start.func1(0x0, 0xc0004dad40)
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/pkg/scanners/terraform/executor/pool.go:274 +0x2a
github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*Worker).Start(0xc0004dad40)
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/pkg/scanners/terraform/executor/pool.go:279 +0xb6
created by github.com/aquasecurity/defsec/pkg/scanners/terraform/executor.(*Pool).Run
	/root/go/pkg/mod/github.com/aquasecurity/defsec@v0.46.3/pkg/scanners/terraform/executor/pool.go:55 +0xef
##[error]Bash exited with code '1'.
Finishing: TFSec Static Code Analysis
```

**System Info**
 - tfsec version: 1.19.1
 - terraform version: 0.15.4 Provider version 3.4.0
 - OS: Ubuntu 20.04.4 LTS

**Example Code**
```hcl
resource "azurerm_storage_account" "storage" {
  account_kind                           = var.sa_kind
  account_replication_type        = var.sa_replication_type
  account_tier                            = var.sa_tier
  location                                   = "North Central US"
  min_tls_version                       = "TLS1_2"
  name                                      = "${var.tenant}${var.subscription}${var.sa_name_short}sta"
  resource_group_name           = var.resource_group_name

  network_rules {
    default_action                     =  var.sa_network_rules_default_action
    ip_rules = [
    ]
  }
}

variable "resource_group_name" {}
variable "sa_kind" {}
variable "sa_name_short" {}
variable "sa_network_rules_default_action" {}
variable "sa_replication_type" {}
variable "sa_tier" {}
variable "subscription" {}
variable "tenant" {}
```
**Variable is set as follows in the env.hcl file:**
 sa_network_rules_default_action                   = "Allow"

**Additional context**
If we hardcode the  "default_action" TFSec works as expected. 
If we set a default for the variable to Allow or Deny, TFSec works as expected.
If a terraform.tfvars file is included in the directory and has a value for the variable, it works if we are setting the switch to use it:
tfsec --force-all-dirs --tfvars-file terraform.tfvars

We would like to see this work to look for values in an env.hcl file that is stored two folders above the deployment in the folder structure.

--- a/internal/adapters/terraform/azure/storage/adapt.go
+++ b/internal/adapters/terraform/azure/storage/adapt.go
@@ -149,6 +149,8 @@ func adaptNetworkRule(resource *terraform.Block) storage.NetworkRule {
 		allowByDefault = types.Bool(true, defaultActionAttr.GetMetadata())
 	} else if defaultActionAttr.Equals("deny", terraform.IgnoreCase) {
 		allowByDefault = types.Bool(false, defaultActionAttr.GetMetadata())
+	} else {
+		allowByDefault = types.BoolDefault(false, resource.GetMetadata())
 	}
 
 	if resource.HasChild("bypass") {

--- a/internal/adapters/terraform/azure/storage/adapt.go
+++ b/internal/adapters/terraform/azure/storage/adapt.go
@@ -1,6 +1,8 @@
 package storage
 
 import (
+	"strings"
+
 	"github.com/aquasecurity/defsec/internal/types"
 	"github.com/aquasecurity/defsec/pkg/providers/azure/storage"
 	"github.com/aquasecurity/defsec/pkg/terraform"
@@ -145,10 +147,14 @@ func adaptNetworkRule(resource *terraform.Block) storage.NetworkRule {
 	var bypass []types.StringValue
 
 	defaultActionAttr := resource.GetAttribute("default_action")
-	if defaultActionAttr.Equals("allow", terraform.IgnoreCase) {
-		allowByDefault = types.Bool(true, defaultActionAttr.GetMetadata())
-	} else if defaultActionAttr.Equals("deny", terraform.IgnoreCase) {
-		allowByDefault = types.Bool(false, defaultActionAttr.GetMetadata())
+
+	if defaultActionAttr.IsNotNil() {
+		switch strings.ToLower(defaultActionAttr.Value().AsString()) {
+		case "allow":
+			allowByDefault = types.Bool(true, defaultActionAttr.GetMetadata())
+		case "deny":
+			allowByDefault = types.Bool(false, defaultActionAttr.GetMetadata())
+		}
 	} else {
 		allowByDefault = types.BoolDefault(false, resource.GetMetadata())
 	}

